pkgs/by-name/im/immich-public-proxy/package.nix

```diff
@@ -0,0 +1,46 @@
+{
+  lib,
+  buildNpmPackage,
+  fetchFromGitHub,
+  nixosTests,
+  nodejs,
+}:
+let
+  pname = "immich-public-proxy";
+  version = "1.5.3+unstable=2024-12-02";
+  repo = fetchFromGitHub {
+    owner = "alangrainger";
+    repo = pname;
+    # 1.5.3 plus upstreamed nixpkgs patches
+    rev = "e4d66a5cccf67bb9c3cf7939b39c4569ba064b35";
+    hash = "sha256-VClwkhuvp6gZNZaTp7gqjHy+4uiGJPJFF/1/Sb4KCRw=";
+  };
+in
+buildNpmPackage {
+  inherit pname version;
+  src = "${repo}/app";
+
+  npmDepsHash = "sha256-dEdEonVS+NFfov+w/wupTtc55ww9zx61n4qFM29mcUY=";
+
+  # patch in absolute nix store paths so the process doesn't need to cwd in $out
+  postInstall = ''
+    sed -i "/const app =/a // Set the views path to the nix output\napp.set('views', '$out/lib/node_modules/immich-public-proxy/views')" \
```

Comment:
Can we use `substituteInPlace`? This should also be done in `postPatch` I think.

Reply:
I had opted for `sed` because I needed to insert rather than replace, and I needed to insert newlines, which `substituteInPlace` didn't seem to respect when I used `"\n"`. It looks like I can use `substituteInPlace` by putting actual newlines in, which is... maybe still better than `sed`?